### DISCUSSION TITLE
Notifications for incoming transactions

### DISCUSSION THREAD

**Discussion by dannyswap** [btcpayserver/btcpayserver#4051] [Ideas / Feature Requests]
Currently notifications are only possible when invoice is settled or paid to expired invoices.
It would be very helpful if you can add notification options for any incoming transactions.

Sometimes client may pay partial payment for the second time into the invoice and there is no way to see that from a guest account. Or a client pays into a settled invoice.

I think its pretty basic to be able to be notified whenever a transaction comes in, it exists on virtually any wallet today.
Backoffice worker who sits on a guest user, money comes in, should be able to immediately click into knowing what invoice that money belongs to.

Thanks!

**Comment by bolatovumar** [comment]
@dannyswap are you referring to this page and would like to see partial and/or unconfirmed transactions in the list?

![Capture](https://user-images.githubusercontent.com/1934678/185729120-d208b76a-f12b-4da9-b6f3-1f11a603ada3.PNG)

**Reply by electrumhelp1** [reply]
> @dannyswap are you referring to this page and would like to see partial and/or unconfirmed transactions in the list?
> 
> ![Capture](https://user-images.githubusercontent.com/1934678/185729120-d208b76a-f12b-4da9-b6f3-1f11a603ada3.PNG)

Yes. Notifications in this page also pop up in the bell icon and are especially handy for Guest users to be notified of events that are relevant in the account.

When there are hundreds of invoices out, there is no indication for Guest user if an invoice has been paid into or in general if money has come into the wallet.

My request is a notification to be sent when any transaction is being paid into the account whether the invoice has already been settled or not. 

Meaning:
1) John has a $10,000 invoice out in invoice #123. John paid $1,000 to the wallet in the invoice. Notification pops that partial payment has been paid.
2) John has a $1,000 invoice out in invoice #456. John has already paid the full invoice on August 1st and it has been settled. However, John decided to send an additional $1,000 on September 1st, using the same wallet as in this invoice #456. Notification pops that a transaction has been paid into an invoice.

Any general notification of incoming funds would do for my request. The issue is mostly targeted to Guest users who have no indication of transactions other than by going into each invoice individually and are only notified when an invoice is Settled or Paid after expiration. 


Thanks

**Reply by pavlenex** [reply]
Would enabling notification for guest user about the invoice be sufficient?

**Reply by electrumhelp1** [reply]
@pavlenex Guest user once inside the platform already gets notifications for settled invoices today (as well as other notifications). However, incoming transaction to an invoice does not notify. 

The request is to receive notifications for incoming transactions so whenever money comes into the wallet he receives a notification and can click it to be taken to the invoice the money was paid into. 

The idea is that John pays into an invoice at any time, any amount, notification pops up "A transaction was made into invoice XXX", you click the notification and it takes you to the relevant invoice where you can see the payment.

**Comment by woutersamaey** [comment]
I want webhooks for all incoming transactions, regardless of invoices. This is kind of similar.
Withdrawals from a custodian account to the store wallet would be like that: an incoming transaction without invoice.
